### PR TITLE
[MLIR][Presburger] Fix reduce bug in Fraction class and add tests

### PR DESCRIPTION
**Pull request by Abhinav271828**
Modified `Fraction.reduce()` to work with negative fractions as well.
Added tests to verify the implementation of arithmetic and relational operators on Fractions.

### CODE REVIEW

**Review by Groverkss — CHANGES_REQUESTED** [review]
Please remove the libcxx changes

File: mlir/unittests/Analysis/Presburger/FractionTest.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  f += g;
+  EXPECT_EQ(f, Fraction(Fraction(1, 12)));
+  f -= g;
+  EXPECT_EQ(f, Fraction(3, 4));
+}
+
+TEST(FractionTest, relational) {
+  Fraction f(2, 5), g(3, 7);
+  ASSERT_TRUE(f < g);
+  ASSERT_FALSE(g < f);
```
Comment (Superty) [inline mlir/unittests/Analysis/Presburger/FractionTest.cpp:48]: You can use EXPECT_TRUE/EXPECT_FALSE here

File: mlir/unittests/Analysis/Presburger/FractionTest.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  EXPECT_EQ(f, Fraction(4, 10));
+}
```
Comment (Superty) [inline mlir/unittests/Analysis/Presburger/FractionTest.cpp:51]: please add a new line to the end of the file

**Review by superty — CHANGES_REQUESTED** [review]

**Comment by Superty** [comment]
Thanks!

**Pull request merged**